arraycontext/impl/pytato/compile.py

```diff
@@ -414,11 +413,10 @@ def _dag_to_transformed_pytato_prg(self, dict_of_named_arrays, *, prg_id=None):
                 prg_id, "pre_generate_loopy", pt_dict_of_named_arrays)
 
         with ProcessLogger(logger, f"generate_loopy for '{prg_id}'"):
+            from arraycontext.loopy import _DEFAULT_LOOPY_OPTIONS
             pytato_program = pt.generate_loopy(
                     pt_dict_of_named_arrays,
-                    options=lp.Options(
-                        return_dict=True,
-                        no_numpy=True),
+                    options=_DEFAULT_LOOPY_OPTIONS,
```

Comment:
`return_dict` is needed for correctness. Could you add an `assert` to ensure it's present?

Reply:
Sure! What do you think of [fc0610a](https://github.com/inducer/arraycontext/pull/238/commits/fc0610ad95355d9ad0e74bd55d7f8997c7f02913)? (also added for `freeze()`)

Reply:
LGTM, thanks!